src/middlewared/middlewared/common/ports/__init__.py

```diff
@@ -27,10 +28,28 @@ async def basic_checks(self):
         elif not isinstance(self.port_fields, Iterable):
             raise ValueError('Port fields must be an iterable')
 
+    def bind_address(self, config):
+        default = '0.0.0.0'
+        return default if self.bind_address_field is NotImplementedError else (
+            config.get(self.bind_address_field) or default
+        )
+
+    def get_bind_ip_port_tuple(self, config, port_field):
+        return self.bind_address(config), config[port_field]
+
+    async def config(self):
+        return await self.middleware.call(f'{self.namespace}.config')
+
+    async def get_ports_internal_override(self):
```

Comment:
Why not give it a more descriptive name that tells us that this method returns the ports that we are listening on all IP addresses?